### PR TITLE
Separate MLeap tests and unpin pyspark

### PR DESCRIPTION
**Pull request by harupy**
## What changes are proposed in this pull request?

Separate MLeap tests and unpin pyspark to use the latest version

## How is this patch tested?

Existing tests

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: tests/pyfunc/test_spark.py [hunk 1]
```diff
@@ -54,11 +54,6 @@ def configure_environment():
 
 
 def get_spark_session(conf):
-    # setting this env variable is needed when using Spark with Arrow >= 0.15.0
-    # because of a change in Arrow IPC format
-    # https://spark.apache.org/docs/latest/sql-pyspark-pandas-with-arrow.html# \
-    # compatibiliy-setting-for-pyarrow--0150-and-spark-23x-24x
-    os.environ["ARROW_PRE_0_15_IPC_FORMAT"] = "1"
```
Comment (harupy) [inline tests/pyfunc/test_spark.py:61]: This causes an error in spark 3.x.

File: tests/pyfunc/test_spark.py [hunk 2]
```diff
@@ -140,7 +135,7 @@ def predict(self, context, model_input):
                 columns=["a", "b", "c", "d"], data={"a": [1], "b": [2], "c": [3], "d": [4]}
             )
         )
-        with pytest.raises(Py4JJavaError):
+        with pytest.raises(pyspark.sql.utils.PythonException):
```
Comment (harupy) [inline tests/pyfunc/test_spark.py:138]: Fix for this failure:

https://github.com/mlflow/mlflow/runs/2330220978#step:6:611

File: tests/pyfunc/test_spark.py [hunk 3]
```diff
@@ -140,7 +135,7 @@ def predict(self, context, model_input):
...
             res = data.withColumn("res1", udf("a", "b")).select("res1").toPandas()
```
Comment (harupy) [inline tests/pyfunc/test_spark.py:139]: Fix for this failure:

https://github.com/mlflow/mlflow/runs/2330220978#step:6:611

```
>               raise converted from None
E               pyspark.sql.utils.PythonException: 
E                 An exception was thrown from the Python worker. Please see the stack trace below.
```

File: dev/extra-ml-requirements.txt [hunk 4]
```diff
@@ -40,7 +40,7 @@ onnxruntime
 # mleap format via ``mlflow.spark.log_model``, ``mlflow.spark.save_model``
 mleap
 # Required by mlflow.spark
-pyspark==2.4.0
+pyspark
```
Comment (WeichenXu123) [inline dev/extra-ml-requirements.txt:43]: pyspark>=2.4.0 ?

File: dev/run-python-sagemaker-tests.sh [hunk 5]
```diff
@@ -20,6 +20,10 @@ pytest --verbose tests/sagemaker/mock --large
 
 # Added here due to dependency on sagemaker
 # TODO: split out sagemaker tests and move other spark tests to run-python-flavor-tests.sh
-pytest --verbose tests/spark --large
+pytest --verbose tests/spark --large  --ignore tests/spark/test_mleap_model_export.py
+
+# MLeap doesn't support spark 3.x (https://github.com/combust/mleap#mleapspark-version)
+pip install pyspark==2.4.5
+pytest --verbose tests/spark/test_mleap_model_export.py --large
```
Comment (WeichenXu123) [inline dev/run-python-sagemaker-tests.sh:27]: Do we need a separate CI action for mleap ?

Comment (harupy) [reply dev/run-python-sagemaker-tests.sh:27]: What is the motivation for that?

File: dev/extra-ml-requirements.txt (hunk 4 above)
Comment (harupy) [reply dev/extra-ml-requirements.txt:43]: I don't think we need to do that.

**Review by WeichenXu123 — APPROVED** [review]

**Pull request merged**